packages/react-select/src/Async.js

```diff
@@ -86,15 +86,17 @@ export const makeAsyncSelect = <C: {}>(
         });
       }
     }
-    UNSAFE_componentWillReceiveProps(nextProps: C & AsyncProps) {
+    componentDidUpdate(prevProps: C & AsyncProps) {
       // if the cacheOptions prop changes, clear the cache
-      if (nextProps.cacheOptions !== this.props.cacheOptions) {
+      if (prevProps.cacheOptions !== this.props.cacheOptions) {
         this.optionsCache = {};
       }
```

Comment:
I don't believe this will work correctly, because `componentDidUpdate` happens after `render` and so it will still render the old options the first time the component is rendered after `cacheOptions` changes.